webapi/src/main/scala/org/knora/webapi/slice/common/ApiComplexV2JsonLdRequestParser.scala

```diff
@@ -43,6 +44,77 @@ final case class ApiComplexV2JsonLdRequestParser(
+        valueIri               <- valueIri(valueResource).someOrFail("The value IRI is required")
+        newValueVersionIri     <- newValueVersionIri(valueResource, valueIri)
+        valueCreationDate      <- ZIO.fromEither(valueCreationDate(valueResource))
+        valuePermissions       <- ZIO.fromEither(valuePermissions(valueResource))
+        valueFileValueFilename <- ZIO.fromEither(valueFileValueFilename(valueResource))
+        valueContent <-
+          getValueContent(valueType.toString, valueResource, valueFileValueFilename, resourceIri.shortcode, ingestState)
+            .map(Some(_))
+            .orElse(ZIO.none)
+        updateValue <- valueContent match
+                         case Some(valueContentV2) =>
+                           ZIO.succeed(
+                             UpdateValueContentV2(
+                               resourceIri.toString,
+                               resourceClassIri.smartIri,
+                               valuePropertyIri.smartIri,
+                               valueIri.toString,
+                               valueContentV2,
+                               valuePermissions,
+                               valueCreationDate,
+                               newValueVersionIri.map(_.smartIri),
+                               ingestState,
+                             ),
+                           )
+                         case None =>
+                           ZIO
+                             .fromOption(valuePermissions)
+                             .mapBoth(
+                               _ => "No permissions and no value content found",
+                               permissions =>
+                                 UpdateValuePermissionsV2(
+                                   resourceIri.toString,
+                                   resourceClassIri.smartIri,
+                                   valuePropertyIri.smartIri,
+                                   valueIri.toString,
+                                   valueType,
+                                   permissions,
+                                   valueCreationDate,
+                                   newValueVersionIri.map(_.smartIri),
+                                 ),
+                             )
+      } yield updateValue
+    }
+
+  private def newValueVersionIri(r: Resource, valueIri: ValueIri): IO[String, Option[ValueIri]] =
+    ZIO
+      .fromEither(r.objectUriOption(NewValueVersionIri))
+      .some
+      .flatMap(converter.asSmartIri(_).mapError(_.getMessage).asSomeError)
+      .flatMap(iri => ZIO.fromEither(ValueIri.from(iri)).asSomeError)
+      .filterOrFail(newV => newV != valueIri)(
+        Some(s"The IRI of a new value version cannot be the same as the IRI of the current version"),
+      )
+      .filterOrFail(newV => newV.sameResourceAs(valueIri))(
+        Some(
+          s"The project shortcode and resource must be equal for the new value version and the current version",
```

Comment:
```suggestion
          s"The project shortcode and resource must be equal to the new value version and the current version.",
```

Reply:
Actually, `for` is correct in this case.

Bot shortcode and resource id must be the same for the two iris new value and current value version iris. The Shortcode and resource id are part of the iris.